src/FileIndexer.ts

```diff
@@ -1,601 +1,2459 @@
-        return aliased
-      }
-    }
-    return symbol
-  }
-
-  private visitIdentifier(identifier: ts.Identifier, sym: ts.Symbol): void {
-    const range = Range.fromNode(identifier).toLsif()
-    let role = 0
-    const isDefinition = this.declarationName(identifier.parent) === identifier
-    if (isDefinition) {
-      role |= lsiftyped.SymbolRole.Definition
-    }
-    for (const declaration of sym?.declarations || []) {
-      const lsifSymbol = this.lsifSymbol(declaration)
-      if (lsifSymbol.isEmpty()) {
-        // Skip empty symbols
-        continue
-      }
-      this.document.occurrences.push(
-        new lsif.lib.codeintel.lsiftyped.Occurrence({
-          range,
-          symbol: lsifSymbol.value,
-          symbol_roles: role,
-        })
-      )
-      if (isDefinition) {
-        this.addSymbolInformation(identifier, sym, declaration, lsifSymbol)
-        this.handleShorthandPropertyDefinition(declaration, range)
-        // Only emit one symbol for definitions sites, see https://github.com/sourcegraph/lsif-typescript/issues/45
-        break
-      }
-    }
-  }
-
-  /**
-   * Handles the special-case around shorthand property syntax so that we emit two occurrences instead of only one.
-   * Shorthand properties need two symbols because they both define a symbol and reference a symbol. For example:
-   * ```
-   * const a = 42
-   * const b = {a}
-   * //         ^ both references the local const `a` and defines a new property
-   * const c = b.a
-   * //          ^ reference to the property `a`, not the local const
-   * ```
-   */
-
-  private handleShorthandPropertyDefinition(
-    declaration: ts.Node,
-    range: number[]
-  ): void {
-    if (declaration.kind === ts.SyntaxKind.ShorthandPropertyAssignment) {
-      const valueSymbol =
-        this.checker.getShorthandAssignmentValueSymbol(declaration)
-      if (!valueSymbol) {
-        return
-      }
-      for (const symbol of valueSymbol?.declarations || []) {
-        const lsifSymbol = this.lsifSymbol(symbol)
-        if (lsifSymbol.isEmpty()) {
-          continue
-        }
-        this.document.occurrences.push(
-          new lsif.lib.codeintel.lsiftyped.Occurrence({
-            range,
-            symbol: lsifSymbol.value,
-          })
+        if (this.dev)
+            console.log(
+                `${this.indent()}• visitDeclarationWithBody [${node.pos}:${node.end}]`,
+                node.kind,
+                operatorsToPush,
+                node
+            );
+        let lsifSymbol = this.lsifSymbol(node);
+
+        if (this.underTest) {
+            this.pushComponentToHeirarchy(this.lsifCounter.next());
+        } else {
+            let id = this.emitDeclaration(node, lsifSymbol);
+            this.pushComponentToHeirarchy(id);
+            this.getAndStoreReferences(id, node); // TODO - uncomment <-----------------------------------------<<<<<<<<<<<<<<<
+        }
+
+        // if (this.underTest) {
```

Comment:
yes - I know.  This was to be thorough.  However, the declarations used to be combined - they were split because of too many errors.  Refactoring can be done in the future, but I had to split it to get it working properly.  We can revisit.